src/Moryx.AbstractionLayer.Products.Endpoints/ProductConverter.cs

```diff
@@ -269,22 +269,27 @@ private void UpdateCollection(IList value, IEnumerable<PartModel> parts)
                 if (partModel is null)
                     continue;
 
-                var match = unused.Find(r => r.Id == partModel?.Id);
-                if (match == null)
+                var oldPartMatch = oldParts.Find(r => r.Id == partModel.Id);
+                // new partlink
+                if (oldPartMatch == null)
                 {
-                    match = (IProductPartLink)Activator.CreateInstance(elemType);
-                    value.Add(match);
+                    oldPartMatch = (IProductPartLink)Activator.CreateInstance(elemType);
+                    oldPartMatch.Product = _productManagement.LoadType(partModel.Product.Id);
+                    value.Add(oldPartMatch);
                 }
+                //modified reference
+                else if (oldPartMatch.Product.Id != partModel.Product.Id)
+                    oldPartMatch.Product = _productManagement.LoadType(partModel.Product.Id);
```

Comment:
Sorry if its a double post: cannot find my posts. 
Remove from oldParts list otherwise changed instance would be deleted at the end. I think this code was never executed otherwise i cannot explain why this never causes problems...

Reply:
@Toxantron, did we ever replace a product of a partlink without creating a new partlink?

Reply:
Good question. I don't think so, but that is only the current definition of the endpoint. There is nothing preventing someone from replacing a reference on an exisiting part link through the facade.

Reply:
This was not fixed: _**Remove from oldParts list otherwise changed instance would be deleted at the end.**_ @MathoMathiasCamara

Reply:
Damn, you are right.